Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -587,15 +619,35 @@ void CGPathAddCurveToPoint(CGMutablePathRef path,
                            CGFloat cp2y,
                            CGFloat x,
                            CGFloat y) {
-    UNIMPLEMENTED();
+    RETURN_IF(!path);
+
+    CGPoint endPoint = CGPointMake(x, y);
+    CGPoint controlPoint1 = CGPointMake(cp1x, cp1y);
+    CGPoint controlPoint2 = CGPointMake(cp2x, cp2y);
+
+    ComPtr<ID2D1PathGeometry> newPath;
+    ComPtr<ID2D1GeometrySink> newSink;
+
+    FAIL_FAST_IF_FAILED(_createPathReadyForFigure(path, &newPath, &newSink, _getInvertedCurrentPointOfPath(path)));
+    newSink->AddBezier(D2D1::BezierSegment(_CGPointToD2D_F(controlPoint1), _CGPointToD2D_F(controlPoint2), _CGPointToD2D_F(endPoint)));
+    newSink->EndFigure(D2D1_FIGURE_END_OPEN);
+    FAIL_FAST_IF_FAILED(newSink->Close());
+
+    FAIL_FAST_IF_FAILED(path->SimplifyGeometryToPathWithTransformation(newPath, transform));
+
+    if (transform) {
+        endPoint = CGPointApplyAffineTransform(endPoint, *transform);
+    }
+    path->SetCurrentPoint(endPoint);
+    path->SetLastTransform(transform);
```

Comment:
A lot of this code is duplicated.  I suggest making a utility function to which you can pass a lambda to draw the figure and then simplify it into the path.